web/src/state_data.ts

```diff
@@ -21,6 +21,13 @@ export const narrow_term_schema = z.object({
 export type NarrowTerm = z.output<typeof narrow_term_schema>;
 // Sync this with zerver.lib.events.do_events_register.
 
+const onboarding_steps_schema = z.object({
+    type: z.string(),
+    name: z.string(),
+});
```

Comment:
How did you determine this is the schema?

Search for `onboarding_steps` on https://zulip.com/api/register-queue and you'll see it has other fields.

Reply:
Yes, I missed it. Actually one time notices has only these two fields in their schema. Also the hotspots has one extra field named location which is not mentioned in the api, i guess it also should be added ( or maybe is it getting added on the frontend).
Will fix it.

Reply:
This type could be written much better as a union or two types, one for each value of `type`; want to give that a try?

Reply:
Yes, I agree. It would be helpful to improve the code readability also. Working on it.